### PR TITLE
fix: Remove unused files; add more tests; fix some Make targets and lint

### PR DESCRIPTION
**Pull request by timmc-edx**
Remove unused files from root of each cookiecutter. These were not actually
used as they were not inside the `{{cookiecutter.repo_name}}` dir that
would lead to them being included.

More information on removals:

- `openedx.yaml` removed from xblock cookiecutter root and
  https://github.com/openedx/edx-cookiecutters/issues/334 filed to cover
  actually adding that tag. (The rest of the file was either redundant
  with what's in python-template, or obsolete.)
- `requirements.txt` was not used in CI or baking
- Makefiles were not used. This is unfortunate because the IDA one actually
  had some useful tests in it. I've moved those to the ci.yml workflow for
  expediency but they should really be put into the unit tests at some
  point so that they can be run locally. A lot of the django-app Makefile
  seemed to have been written while the repo was still in development, and
  relied on packages that are not installed.

For the moved tests to pass, I had to make some fixes to the IDA template:

- Add missing `docs` target to Makefile, although it doesn't quite work yet;
  filed https://github.com/openedx/edx-cookiecutters/issues/339 for fixing
  this later.
- Add some long-line lint suppression where needed

Other improvements:

- Remove unused `BAKE_OPTIONS` from root Makefile
- Use short version of `BROWSER` script in django-ida Makefile to
  match other Makefiles
- Fix root Makefile so that it actually uses pip-sync rather than just
  installing it
- Update `JWT_PUBLIC_SIGNING_JWK_SET` to be readable and copyable as JSON
  (as is likely to be done for edx-platform in the ongoing JWK work)
- Made `fake_translations` Makefile target actually do something.

**Merge checklist:**
Check off if complete *or* not applicable:
- [x] Changelog record added
- [x] Documentation updated (not only docstrings)
- [x] Fixup commits are squashed away
- [x] Unit tests added/updated
- [x] Manual testing instructions provided
- [x] Noted any: Concerns, dependencies, deadlines, tickets

### CODE REVIEW

File: cookiecutter-django-ida/{{cookiecutter.repo_name}}/Makefile [hunk 1]
```diff
@@ -32,6 +32,10 @@ clean: ## delete generated byte code and coverage reports
 	rm -rf assets
 	rm -rf pii_report
 
+docs: ## generate Sphinx HTML documentation, including API docs
+	tox -e docs
+	$(BROWSER)docs/_build/html/index.html
```
Comment (rgraber) [inline cookiecutter-django-ida/{{cookiecutter.repo_name}}/Makefile:37]: [nit] should there be a space here after $(BROWSER)?

**Review by rgraber — COMMENTED** [review]
One question

File: cookiecutter-django-ida/{{cookiecutter.repo_name}}/Makefile (hunk 1 above)
Comment (timmc-edx) [reply cookiecutter-django-ida/{{cookiecutter.repo_name}}/Makefile:37]: Interestingly enough, no! ...but only if the correct `BROWSER` is defined above, which was not the case. There's a different, more complicated version used in about 40% of our Makefiles that does require a space. However, the other Makefiles in this repo use the short, no-space version, so I'll switch it to that. Good catch!

**Comment by timmc-edx** [comment]
`make docs` is now failing. But guess what: The only reason it wasn't failing previously is that I hadn't added `docs` to the `.PHONY` list, so it was just saying `make: 'docs' is up to date.` and not actually building the docs. -.- Looking into it.

File: .github/workflows/ci.yml [hunk 2]
```diff
@@ -38,3 +41,67 @@ jobs:
...
+      run: |
+        source .venv/bin/activate
+        make validation_requirements
+        make validate
+
+    - name: "Ensure translations can be compiled"
+      working-directory: repo_name
+      run: |
+        source .venv/bin/activate
+        make fake_translations
```
Comment (timmc-edx) [inline .github/workflows/ci.yml:94]: Turns out this make target is just... empty? :-/

**Comment by timmc-edx** [comment]
I think `make docs` is failing in the IDA output because the `cookiecutter.requires_django` variable isn't being respected. This means that the "Configure Django for autodoc usage" block isn't included, so importing urls.py fails.

File: .github/workflows/ci.yml (hunk 2 above)
Comment (rgraber) [reply .github/workflows/ci.yml:94]: If it's empty should we remove this?

File: CHANGELOG.rst [hunk 3]
```diff
@@ -5,6 +5,24 @@ Change Log
...
+
+Changed
+=======
+
+- Use short version of ``BROWSER`` script in django-ida Makefile to match others
+
+Added
+=====
+
+- Improve testing for ``cookiecutter-django-ida`` (migrations, quality check, docker image build, docs build)
```
Comment (rgraber) [inline CHANGELOG.rst:24]: no docs build :(

File: CHANGELOG.rst [hunk 4]
```diff
@@ -5,6 +5,24 @@ Change Log
...
+
+Fixed
+=====
+
+- Add missing ``docs`` Makefile target to ``cookiecutter-django-ida`` and suppressed long-line lint as appropriate
+
+Changed
+=======
+
+- Use short version of ``BROWSER`` script in django-ida Makefile to match others
```
Comment (rgraber) [inline CHANGELOG.rst:19]: Might also add "removed unused files in django-ida, django-app, and xblock"

**Review by rgraber — COMMENTED** [review]
Questions and suggestions for changelog

File: CHANGELOG.rst (hunk 3 above)
Comment (timmc-edx) [reply CHANGELOG.rst]: Removed :-(

File: CHANGELOG.rst (hunk 4 above)
Comment (timmc-edx) [reply CHANGELOG.rst:20]: Added something like this

File: .github/workflows/ci.yml (hunk 2 above)
Comment (timmc-edx) [reply .github/workflows/ci.yml:94]: I've fixed this target.

**Comment by timmc-edx** [comment]
OK, got one last thing squared away. Ready for re-review.

**Review by rgraber — APPROVED** [review]
LGTM

**Pull request merged**